### ISSUE TITLE
Volumes do not get dropped when removing the docker container

### ISSUE THREAD

**Issue by zentavr** [ansible/ansible-modules-core#4057]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

_docker module
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### CONFIGURATION

```
[defaults]
host_key_checking = False
forks = 10
var_compression_level=9
retries=5
fact_caching = jsonfile
fact_caching_connection = ~/.ansible/tmp
```
##### OS / ENVIRONMENT

OS From: OS X, Ubuntu 14.04
OS To: Ubuntu 14.04
##### SUMMARY

Volumes do not get dropped when removing the docker container
##### STEPS TO REPRODUCE
1. Create the docker container with attached volumes
2. fill the volume
3. drop the container
4. repeat several times
5. `docker volume ls -f dangling=true` will show you the not-used containers
   See [Docker manual](https://docs.docker.com/v1.10/engine/userguide/containers/dockervolumes/):

> Note: Docker will not warn you when removing a container without providing the `-v` option to delete its volumes. If you remove containers without using the `-v` option, you may end up with “dangling” volumes; volumes that are no longer referenced by a container. You can use `docker volume ls -f dangling=true` to find dangling volumes, and use `docker volume rm <volume name>` to remove a volume that’s no longer needed.

I spin up the container with something like this:

```
- name: Spinning up an {{ persistcomponent }}-platform container
  docker:
    registry: "{{ docker_registry_url }}"
    email: "{{ docker_registry_email }}"
    username: "{{ docker_registry_user }}"
    password: "{{ docker_registry_password }}"
    docker_api_version: "{{ docker_api_version }}"
    image: "{{ docker_registry_url }}/arena/{{ docker_images[persistcomponent] }}:{{ builds['platform'][persistcomponent] }}"
    pull: always
    state: reloaded
    restart_policy: always
    restart_policy_retry: 5
    name: "{{ persistcomponent }}-platform"
    net: bridge
    volumes:
      - /srv/gsn/arenasettings.json:/srv/gsn/arenasettings.json:ro
    log_driver: fluentd
    log_opt:
      fluentd-address: 127.0.0.1:24224
      fluentd-tag: "docker.{{ '{{' }}.Name{{ '}}' }}"
      fluentd-async-connect: "true"
    #log_driver: syslog
    #log_opt:
    #  syslog-address: udp://127.0.0.1:15140
    #  syslog-tag: "docker.{{ '{{' }}.Name{{ '}}' }}"
    env:
      ARENA_COMPONENT: "{{ persistcomponent }}"
  tags:
    - "{{ persistcomponent }}_platform_spinup"
    - platform_spinup
    - "{{ persistcomponent }}_platform"
  when: builds['platform'][persistcomponent] not in [ 'Keep', 'None']
  notify:
    - "drop {{ persistcomponent }}-nginx container"
    - "restart {{ persistcomponent }}-nginx container"
```

The container has `VOLUME` in its Dockerfile which we mount to nginx container (we put static files there, collected with Django's `manage.py collectstatic`
##### EXPECTED RESULTS

not to have those dangling volumes
##### ACTUAL RESULTS

we have dangling volumes when we renew the container

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by jctanner** [comment]
duplicate_of https://github.com/ansible/ansible/issues/19858
